### DISCUSSION TITLE
Store data in MongoDB

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8555] [Q&A]
Hello, I have a use case and I need your view on how I could do that.
In jobs, we are producing dict  / json data / and we want to store them in MongoDB.

Pure monogo is:

```python
from pymongo import MongoClient

class Connect(object):
    @staticmethod    
    def get_connection():
        return MongoClient("mongodb://$[username]:$[password]@$[hostlist]/$[database]?authSource=$[authSource]")

from connect import Connect
from pymongo import MongoClient

connection = Connect.get_connection()
db = client.test
db.inventory.insert_one(
    {"item": "canvas",
     "qty": 100,
     "tags": ["cotton"],```
     "size": {"h": 28, "w": 35.5, "uom": "cm"}})
``````
and dagster could be:
```python
def my_ops() -&gt; dict:
    return {"item": "canvas",
     "qty": 100,
     "tags": ["cotton"],
     "size": {"h": 28, "w": 35.5, "uom": "cm"}}
```
I think of [ressources](https://docs.dagster.io/concepts/resources) to do that but I am not quite sure on how to start ?

At the end, the mongodb will serve another service. What do you think ?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1638800813439600?thread_ts=1638800813.439600&cid=C01U954MEER).*

**Comment by yuhan** [comment]
You could create a mongo resource which would be available to any op that requires it.  Something like this:
```python
@resource
def mongo_test(context):
    client = MongoClient(...)
    return client.test

@resource
def mongo_prod(context):
    client = MongoClient(...)
    return client.prod

@op(required_resource_keys={"mongo"})
def my_op(context):
    context.resources.mongo.inventory.insert_one({"item": "canvas", ...})

@job(resource_defs={"mongo": test_mongo_resource})
def test_job():
    my_op()
```
Written this way, `my_op` doesn’t need to know if the mongo resource is the test db or the prod db. that’s all set up at the `job` layer, which determines how the set of ops get executed.

**Reply by yuhan** [reply]
Full working example: https://gist.github.com/slamer59/b14a948cd415de2ec29ba33f534d3a1e (credited to @slamer59)
